lib/container_server.go

```diff
@@ -53,7 +54,41 @@ func (c *ContainerServer) Runtime() *oci.Runtime {
 
 // Hooks returns the oci hooks for the ContainerServer
 func (c *ContainerServer) Hooks() map[string]HookParams {
-	return c.hooks
+	hooks := map[string]HookParams{}
+	c.hooksLock.Lock()
+	defer c.hooksLock.Unlock()
+	for key, h := range c.hooks {
+		hooks[key] = h
+	}
+	return hooks
+}
+
+// RemoveHook removes an hook by name
+func (c *ContainerServer) RemoveHook(hook string) {
+	c.hooksLock.Lock()
+	defer c.hooksLock.Unlock()
+	if _, ok := c.hooks[hook]; ok {
+		delete(c.hooks, hook)
+	}
+}
+
+// AddHook adds an hook by hook's path
+func (c *ContainerServer) AddHook(hookPath string) {
+	hook, err := readHook(hookPath)
```

Comment:
Any concerns about this hook being read and then updated/deleted by someone else before you get the lock and then reference it below?